meta-mbl-distro/scripts/lib/update/testinfo.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import hashlib
```

Comment:
TBH I don't think this test info stuff belongs in the `create-update-payloads` it feels like we're encoding too much knowledge of various tests we're going to run in our test farm in what is supposed to be a user-facing tool for generating update payloads. Could we not just have a separate script that generates this "testinfo" stuff? Things like the sha256 of the image can actually be used by `swupdate` for verification so maybe that belongs here.

Reply:
I've been thinking about this for a few days, and I've spoken to @jjohnson-arm about it (I meant to speak to you and @diegorusso about it too but I forgot that you'd be on holiday).

My position is this:
Emitting some info to aid testing is very useful, and generating it in a separate script would have issues:
  * If we used a separate script that used the BitBake environment then there's some chance that the two scripts will disagree about what's in the payload. 
  * If we had a separate script that interrogated an actual payload to work out what's in it then that script would have to deal with unpacking a payload, which would be a bunch of unnecessary effort.

So I definitely think we should have this script creating some information to help with testing. 

In principle, I do think the test info that's created is too prescriptive about what testing should be done - it should be providing information about the payload that will help with testing rather than specifying the tests that should run and IIUC, you, @jjohnson-arm and I agree on this.

I'm not convinced that at this point it would be pragmatic to change the approach though - it would make the testing code more complicated and I don't think it would be worth it.

Reply:
My suggestion would've been to use the BitBake environment and write a new script, which calls the library code you've created, to find the payload images and create the test info from them. I think that would've lowered the probabilty of the two scripts disagreeing over the payload content? It would also remove the test info generation from the `create-update-payload` interface, without having to re-implement the logic from the `create-update-payload` script.

I won't gate the PR on this as it probably needs some more thought, but I do still think we should look into splitting out the test info from the `create-update-payload` script at some point.